### ISSUE TITLE
[FEAT EXTRACTION] Perform slice-by-slice (2D) extraction from 3D ROI mask

### ISSUE THREAD

**Issue by rcuocolo** [AIM-Harvard/pyradiomics#741]
Is it possible through the parameter file settings to replicate what is reported in this [paper](https://academic.oup.com/bjs/article/106/13/1800/6121045#226844266)'s [supplementary materials](https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/bjs/106/13/10.1002_bjs.11410/1/bjs11410-sup-0002-appendix.pdf?Expires=1645197133&Signature=dBn9VNkRSemjC-44-RVtRPJ3wRbNs7Cg0d7CwnUj4BXgcfGxTKvqa4tHykfUFgL2WBtGo8WXnlNdrMqMPZ7WhWVYQUKwQ5YCzyoDdQ61ZyswYoG1DsXsahRVucVQL3Mgbiz98fGtIseTBO2EN2QH-CEAZrTRWHUPdd4dZcppRJ7CsUzTUoEWsRyOYd9mrL9kl3cdGVsP~huMo9Tu~tM90TBlENqPiF3ec1nX~RD2J09rAQJrSTu0oKFCnbi25zl2klbpD6HhYSelIH-S3WX6ZfE73CnShkTv7EQxBJb-XM5inx3IDXrAQ7C2ckjudLajPvfk-csxD1i2fKUWhj7Vgw__&Key-Pair-Id=APKAIE5G5CRDK6RD3PGA)?
In other words, for datasets with high variability on the Z axis, does it make sense to "average" 2D features across the lesion slices to obtain a final global value better representative for the entire lesion? If so, is there a way to obtain this data from PyRadiomics settings, or should each slice be employed as a distinct mask and averaged after the extraction?
I am not entirely sure of the appropriateness of the method and if it could be easily implemented.
Thanks in advance for any help.

**Comment by JoostJM** [comment]
IBSI mentions the option of calculating features on each slice, and then averaging the calculated feature values across slices to obtain the final value. This is not supported in PyRadiomics.

This is because this means that slices with few (or just one) voxels have as much weight in the final feature value as slices with many voxels. What PyRadiomics does support is "force2Dextraction". This computes a single feature value for all slices, but does so without using offsets that move in-between slices. This only affects texture features. In this way, you can allow for a large variation in Z axis, but still compute valid feature values for the entire ROI.

If you want to force the IBSI option, the only option I see would be to indeed make different ROIs for each slice and then computing the averages outside of the PyRadiomics workflow.

**Comment by rcuocolo** [comment]
Thank you for the clear answer. I did not know averaging features was a viable option, I'll keep it in mind as an alternative. For now I will use the 2D extraction option first to address the Z-axis variability problem.
